src/overlays/actors/ovl_Obj_Switch/z_obj_switch.h

```diff
@@ -17,26 +17,16 @@ typedef enum {
 } ObjSwitchType;
 
 typedef enum {
-    /* 0 */ OBJSWITCH_SUBTYPE_FLOOR_0,
-    /* 1 */ OBJSWITCH_SUBTYPE_FLOOR_1,
-    /* 2 */ OBJSWITCH_SUBTYPE_FLOOR_2,
-    /* 3 */ OBJSWITCH_SUBTYPE_FLOOR_3
-} ObjSwitchSubTypeFloor;
-
-typedef enum {
-    /* 0 */ OBJSWITCH_SUBTYPE_EYE_0,
-    /* 1 */ OBJSWITCH_SUBTYPE_EYE_1
-} ObjSwitchSubTypeEye;
-
-typedef enum {
-    /* 0 */ OBJSWITCH_SUBTYPE_CRYSTAL_0,
-    /* 1 */ OBJSWITCH_SUBTYPE_CRYSTAL_1,
-    /* 4 */ OBJSWITCH_SUBTYPE_CRYSTAL_4 = 4
-} ObjSwitchSubTypeCrystal;
+    /* 0 */ OBJSWITCH_SUBTYPE_DEFAULT,      // Normal Yellow Switches
+    /* 1 */ OBJSWITCH_SUBTYPE_TOGGLE,       // Switches that can be turned on and off
+    /* 2 */ OBJSWITCH_SUBTYPE_RESET_OFF,    // Floor Types only, reset itself when not stood on (off by default)
+    /* 3 */ OBJSWITCH_SUBTYPE_RESET_ON,     // Floor Types only, reset itself when not stood on (on by default)
```

Comment:
```suggestion
    /* 2 */ OBJSWITCH_SUBTYPE_RESET_OFF, // Floor Types only, released when not stood on (flag cleared when released)
    /* 3 */ OBJSWITCH_SUBTYPE_RESET_ON,  // Floor Types only, released when not stood on (flag set when released)
```

Reply:
The flag is cleared for both reset subtypes when it's released, the only difference between them is the default state of the switch when you load the room

Reply:
are you sure? I don't think the switch flag is set on init, and either `ObjSwitch_SetOn`/Off is called in `ObjSwitch_FloorUp`/Down depending on the subtype

Reply:
The switch flag doesn't change with that subtype, only the state of the switch itself is affected

It's probably meant to be used to avoid watching a cutscene when you load a room that has a switch with the reset subtype with something on it like a small crate, I don't think it's used in the game though

Reply:
`ObjSwitch_SetOn` sets the switch flag and `ObjSwitch_SetOff` clears it

Reply:
(Resolved in https://github.com/zeldaret/oot/pull/1082/commits/ab983bf2c1966fe506f3230aaf692cfd0d096a03)